spec/core/ics-026-routing-module/README.md

```diff
@@ -610,14 +632,14 @@ interface PacketRecv {
 
 ```typescript
 function handlePacketRecv(datagram: PacketRecv) {
-    module = lookupModule(datagram.packet.destPort)
-    acknowledgement = module.onRecvPacket(datagram.packet)
-    handler.recvPacket(
-      datagram.packet,
-      datagram.proof,
-      datagram.proofHeight,
-      acknowledgement
-    )
+  module = lookupModule(datagram.packet.destPort)
+  handler.recvPacket(
+    datagram.packet,
+    datagram.proof,
+    datagram.proofHeight,
+    acknowledgement
+  )
+  acknowledgement = module.onRecvPacket(datagram.packet)
```

Comment:
acknowledgement is nil on the recvPacket call.

We need to use WriteAcknowledgement after the callback

Reply:
Fixed! Thank you, @AdityaSripal!